client/src/cmdlfem4x50.c

```diff
@@ -0,0 +1,720 @@
+    
+    // status line
+    sprintf(pstring, "  reading ");
+    strcat(string, pstring);
+
+    if (!bsuccess) {
+        
+        sprintf(pstring, _RED_("failed"));
+        strcat(string, pstring);
+
+    } else {
+            
+        sprintf(pstring, _GREEN_("successful "));
+        strcat(string, pstring);
+
+        if (blogin) {
+
+            if (bpwd_given) {
+
+                sprintf(pstring, "(login with 0x%02x%02x%02x%02x)",
+                                    etd->password[0], etd->password[1],
+                                    etd->password[2], etd->password[3]);
+                strcat(string, pstring);
+            
+            } else {
+
+                sprintf(pstring, "(login with default password 0x00000000)");
+                strcat(string, pstring);
+
+            }
+
+        } else {
+
+            if (bpwd_given) {
+
+                sprintf(pstring, "(login failed)");
+                strcat(string, pstring);
+            
+            } else {
+
+                sprintf(pstring, "(no login)");
+                strcat(string, pstring);
+
+            }
+        }
+
+    }
+
+    PrintAndLogEx(NORMAL,"%s\n", string);
+}
+
+int CmdEM4x50Info(const char *Cmd) {
+
+    // envoke reading of a EM4x50 tag which has to be on the antenna because
+    // decoding is done by the device (not on client side)
+
+    bool errors = false, verbose = false;
+    char password[20] = {0x00}, tmpbuff[3];
+    uint8_t cmdp = 0;
+    em4x50_data_t etd;
+    PacketResponseNG resp;
+
+    // init
+    etd.pwd_given = false;
+    
+    while (param_getchar(Cmd, cmdp) != 0x00 && !errors) {
+        switch (tolower(param_getchar(Cmd, cmdp))) {
+
+            case 'h':
+                return usage_lf_em4x50_info();
+
+            case 'p':
+                param_getstr(Cmd, cmdp + 1, password, sizeof(password));
+                
+                // validation
+                if (strlen(password) != 8) {
+    
+                    PrintAndLogEx(WARNING, "\n  error, password has to be 4 bytes\n");
+                    errors = true;
+                    break;
+                }
+                
+                // prepare given password
+                etd.pwd_given = true;
+                for (int i = 0; i < 4; i++)
+                    sscanf(strncpy(tmpbuff, password+2*i, 2), "%2hhx", &etd.password[i]);
+                    
+                cmdp += 2;
+                break;
+
+            case 'v':
+                verbose = true;
+                cmdp += 1;
+                break;
+                
+            default:
+                PrintAndLogEx(WARNING, "  Unknown parameter '%c'", param_getchar(Cmd, cmdp));
+                errors = true;
+                break;
+        }
+    }
+
+    // validation
+     if (errors) {
+         return usage_lf_em4x50_info();
+    }
+
+    // call info command
+    clearCommandBuffer();
+    SendCommandMIX(CMD_LF_EM4X50_INFO, 0, 0, 0, &etd, sizeof(etd));
```

Comment:
OLD/MIX  shouldn't be use anymore,   use  NG format instead.

Reply:
done